### PR TITLE
Feature auth email sign in

### PR DESCRIPTION
**Pull request by ChingPingYang**

### CODE REVIEW

**Comment by taishikato** [comment]
I think we use `Google sign in` not `email & password`?

**Comment by ChingPingYang** [comment]
> I think we will use Google sign in?

Yeah~ I just did it since we have the login-form from Macus( I was exploring firebase auth as well haha). Anyways, the Google log-in is also working. So if we decide to delete the login form later. We still have the Google login

**Comment by taishikato** [comment]
@ChingPingYang thanks we need google sign-in to get displayName data 👍

**Comment by ChingPingYang** [comment]
they can both get displayName now :) I store the userName in firebase. Maybe we can talk about this in the class to see if we wanna keep it or not?

**Comment by taishikato** [comment]
I see thanks!!

File: src/components/SigninModal/MainContent/SigninForm/SigninForm.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                               value={user.password} 
+                               />
+                    </InputContainer>
+                </InputField>
+                <InputField>
+                    <InputLabelContainer htmlFor="rePassword">
+                        <InputLabelText>Re-ender Password</InputLabelText>
+                    </InputLabelContainer>
+                    <InputContainer>
+                        <Input type="rePassword"
```
Comment (taishikato) [inline src/components/SigninModal/MainContent/SigninForm/SigninForm.jsx]: there is no such type `rePassword`. should be `password` 👍 
https://www.w3schools.com/html/html_form_input_types.asp

File: src/components/SigninModal/MainContent/SigninForm/SigninForm.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    const handleSignIn = async (e) => {
+        e.preventDefault();
+        try {
+            await firebase.firestore().collection('users').add({
+                    userName: user.userName,
+                    email: user.email
+                })
+            await firebase.auth().createUserWithEmailAndPassword(user.email, user.password);
```
Comment (taishikato) [inline src/components/SigninModal/MainContent/SigninForm/SigninForm.jsx]: I reckon we should auth the user then save info on DB.
If auth fails we don't have to save that info on DB.
So we want to save the data of only the authed user.
↓ is the better order I think.
```javascript
// ① Auth
await firebase.auth().createUserWithEmailAndPassword(user.email, user.password);

// ② Save in DB
await firebase.firestore().collection('users').add({
  userName: user.userName,
  email: user.email
})
```

File: src/contexts/AuthContext.js [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+            try {   
+                const data = await firebase.firestore().collection('users').get();
+                data.forEach(doc =>{ 
+                    if(user.email === doc.data().email) {   
+                        setAuth({
+                            ...user,
+                            displayName: doc.data().userName
+                        })
+                    }
+                });
```
Comment (taishikato) [inline src/contexts/AuthContext.js]: You can use `where`
```javascript
const data = await firebase
  .firestore()
  .collection('users')
  .where('email', '==', user.email)
  .get();
const userData = data.docs[0].data()
setAuth({
  ...user,
   displayName: userData.userName
})

File: src/App.js [hunk 4]
```diff
@@ -11,7 +12,7 @@ import firebase from './plugins/firebase';
 import { AuthContext } from './contexts/AuthContext';
 
 function App() {
-  const { auth, login } = useContext(AuthContext);
+  const { login } = useContext(AuthContext);
   useEffect(()=>{
     firebase.auth().onAuthStateChanged( user => {
```
Comment (taishikato) [inline src/App.js]: No need space before `user`

**Review by taishikato — APPROVED** [review]

**Pull request merged**